plugin/kubernetes/object/service.go

```diff
@@ -62,7 +62,12 @@ func toService(skipCleanup bool, svc *api.Service) *Service {
 
 	li := copy(s.ExternalIPs, svc.Spec.ExternalIPs)
 	for i, lb := range svc.Status.LoadBalancer.Ingress {
-		s.ExternalIPs[li+i] = lb.IP
+		if &lb.IP != nil {
+			s.ExternalIPs[li+i] = lb.IP
+		} else if &lb.Hostname != nil {
+			s.ExternalIPs[li+i] = lb.Hostname
+		}
+
```

Comment:
pretty sure that the dereferenced string would never be nil.

```suggestion
		if lb.IP != "" {
			s.ExternalIPs[li+i] = lb.IP
			continue
		}
		s.ExternalIPs[li+i] = lb.Hostname
```

Reply:
yep, not sure what i was doing here. fixed now